public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/url_builder.ts

```diff
@@ -7,20 +7,16 @@ export default class UrlBuilder {
     resourceName: string,
     apiVersion: string
   ) {
-    if ((metricDefinition.match(/\//g) || []).length > 1) {
-      const rn = resourceName.split('/');
-      const service = metricDefinition.substring(metricDefinition.lastIndexOf('/') + 1);
-      const md = metricDefinition.substring(0, metricDefinition.lastIndexOf('/'));
-      return (
-        `${baseUrl}/${subscriptionId}/resourceGroups/${resourceGroup}/providers/${md}/${rn[0]}/${service}/${rn[1]}` +
-        `/providers/microsoft.insights/metricNamespaces?api-version=${apiVersion}`
-      );
+    var metricDefinitionArray = metricDefinition.split('/');
+    const resourceNameArray = resourceName.split('/');
+    const provider = metricDefinitionArray.shift();
+    var urlArray = [baseUrl, subscriptionId, 'resourceGroups', resourceGroup, 'providers', provider];
+    for (var i in metricDefinitionArray) {
+      urlArray.push(metricDefinitionArray[i]);
+      urlArray.push(resourceNameArray[i]);
```

Comment:
On first glance this seems like an opportunuty for bugs - this relies on the resource name and metric namespace having the same number of 'levels', otherwise either we'll get undefined in the URL or miss out on segments. 

in practice this might be fine though - not sure what to do about this...

Reply:
Yep, 100% agree and was sort of included in my comment about input validation. I think this same issue applies with the current code though, not that it's an excuse but I was just not sure how far to go with this :)

Reply:
This is the only thing outstanding, I guess I would just check for equality of size and throw an error if they don't match? I guess we would want to assert that the size should be at least 1, not sure if an upper bound matters.